packages/sdk/src/functions/crossChain.ts

```diff
@@ -0,0 +1,58 @@
+import { encodeFunctionData, parseAbi } from 'viem'
+import { ERC_6551_DEFAULT } from '../constants'
+
+type CrossChainCallParams = {
+  to: `0x${string}`
+  value: bigint
+  data: `0x${string}`
+  originChainId: number
+  destinationChainId: number
+}
+
+type CrossChainCallData = {
```

Comment:
Little optimization opportunity here: the footprint of this type is identical to the calldata return types from

**viemV2.ts**
prepareCreateAccount
prepareExecuteCall

**viemV3.ts**
prepareCreateTokenboundV3Account
prepareTokenboundV3Execute

```
  to: `0x${string}`
  value: bigint
  data: `0x${string}`
```

Maybe we genericize the type as `CallData` and throw it in the shared `/types` dir?

Reply:
Then we could extend it for crosschain calls, like so:

```
type CrossChainCallParams = CallData & {
  originChainId: number
  destinationChainId: number
}
```

Reply:
Nice, will do